schema/pgsql/upgrades/025.sql

```diff
@@ -0,0 +1,10 @@
+ALTER TABLE incident ALTER COLUMN severity SET DEFAULT 'ok';
+ALTER TABLE incident_contact ALTER COLUMN role SET DEFAULT 'recipient';
+ALTER TABLE incident_history ALTER COLUMN type SET DEFAULT 'opened';
+
+ALTER TABLE available_channel_type ALTER COLUMN type TYPE varchar(255);
+ALTER TABLE channel ALTER COLUMN type TYPE varchar(255);
+ALTER TABLE contact ALTER COLUMN username TYPE varchar(254);
+ALTER TABLE contact_address ALTER COLUMN type TYPE varchar(255);
+ALTER TABLE object_id_tag ALTER COLUMN tag TYPE varchar(255);
+ALTER TABLE object_extra_tag ALTER COLUMN tag TYPE varchar(255);
```

Comment:
> How did you choose the allowed length, in particular, why is one column shorter than the rest?

Icinga Web user names are <=254. The others... 🤷‍♂️

> Also, when adding constraints to the allowed values, these have to be added to the corresponding input validation functions as well.

Honestly speaking, I'm new here and have no idea what you're talking about. Or do you mean the Web part which writes to the DB via forms?

Reply:
The code that writes to these tables would now receive an error for what worked previously. Given that these were of type `text` before, there are probably no length checks so far.

- For `available_channel_type.type` this affects this function: https://github.com/Icinga/icinga-notifications/blob/32bf26e3217a64697faab1ce9f8497a00e0a9517/internal/channel/plugin.go#L224-L234
- For `object_id_tag.tag` and `object_extra_tag.tag` this affects this function: https://github.com/Icinga/icinga-notifications/blob/32bf26e3217a64697faab1ce9f8497a00e0a9517/internal/event/event.go#L64-L108
- The remaining three columns are written by Web. I guess that might benefit from some "this value is too long" form validation instead of running into a database error.

Reply:
Actually two. One of the three you likely mean is already a FK, hence a selector – not a text box. But yes, ...

* https://github.com/Icinga/icinga-notifications-web/pull/221